test/threeid-provider.test.js

```diff
@@ -160,7 +161,7 @@ describe('ThreeIdProvider', () => {
       expect(idw1.id).toEqual(idw2.id)
     })
 
-    it('Removes authSecret from keychain and creates instance', async () => {
+    it.only('Removes authSecret from keychain and creates instance', async () => {
```

Comment:
@ukstv may be an interesting bug here to look at, added comments here and logs where the update is happening, this fails more often than not, although some times passes. Sometimes it fails on first update, sometimes second. Failed update just means it awaits doc change, but returns with out change being applied. 

Started to try try to replicate in a more limited context - https://github.com/zachferland/doc-change-race/blob/master/index.js but have not yet (this passes fine so far, as expected i guess)